homeassistant/components/rpi_camera/camera.py

```diff
@@ -5,82 +5,44 @@
-CONF_IMAGE_WIDTH = "image_width"
-CONF_TIMELAPSE = "timelapse"
-CONF_VERTICAL_FLIP = "vertical_flip"
-
-DEFAULT_HORIZONTAL_FLIP = 0
-DEFAULT_IMAGE_HEIGHT = 480
-DEFAULT_IMAGE_QUALITY = 7
-DEFAULT_IMAGE_ROTATION = 0
-DEFAULT_IMAGE_WIDTH = 640
-DEFAULT_NAME = "Raspberry Pi Camera"
-DEFAULT_TIMELAPSE = 1000
-DEFAULT_VERTICAL_FLIP = 0
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_FILE_PATH): cv.isfile,
-        vol.Optional(CONF_HORIZONTAL_FLIP, default=DEFAULT_HORIZONTAL_FLIP): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=1)
-        ),
-        vol.Optional(CONF_IMAGE_HEIGHT, default=DEFAULT_IMAGE_HEIGHT): vol.Coerce(int),
-        vol.Optional(CONF_IMAGE_QUALITY, default=DEFAULT_IMAGE_QUALITY): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=100)
-        ),
-        vol.Optional(CONF_IMAGE_ROTATION, default=DEFAULT_IMAGE_ROTATION): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=359)
-        ),
-        vol.Optional(CONF_IMAGE_WIDTH, default=DEFAULT_IMAGE_WIDTH): vol.Coerce(int),
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_TIMELAPSE, default=1000): vol.Coerce(int),
-        vol.Optional(CONF_VERTICAL_FLIP, default=DEFAULT_VERTICAL_FLIP): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=1)
-        ),
-    }
+from .const import (
+    CONF_HORIZONTAL_FLIP,
+    CONF_IMAGE_HEIGHT,
+    CONF_IMAGE_QUALITY,
+    CONF_IMAGE_ROTATION,
+    CONF_IMAGE_WIDTH,
+    CONF_OVERLAY_METADATA,
+    CONF_OVERLAY_TIMESTAMP,
+    CONF_TIMELAPSE,
+    CONF_VERTICAL_FLIP,
+    DOMAIN,
 )
 
+_LOGGER = logging.getLogger(__name__)
 
 def kill_raspistill(*args):
     """Kill any previously running raspistill process.."""
+    _LOGGER.debug("killall raspistill")
     subprocess.Popen(
         ["killall", "raspistill"], stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT
     )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Raspberry Camera."""
+    _LOGGER.debug("setup_platform")
     if shutil.which("raspistill") is None:
         _LOGGER.error("'raspistill' was not found")
         return False
 
-    setup_config = {
-        CONF_NAME: config.get(CONF_NAME),
-        CONF_IMAGE_WIDTH: config.get(CONF_IMAGE_WIDTH),
-        CONF_IMAGE_HEIGHT: config.get(CONF_IMAGE_HEIGHT),
-        CONF_IMAGE_QUALITY: config.get(CONF_IMAGE_QUALITY),
-        CONF_IMAGE_ROTATION: config.get(CONF_IMAGE_ROTATION),
-        CONF_TIMELAPSE: config.get(CONF_TIMELAPSE),
-        CONF_HORIZONTAL_FLIP: config.get(CONF_HORIZONTAL_FLIP),
-        CONF_VERTICAL_FLIP: config.get(CONF_VERTICAL_FLIP),
-        CONF_FILE_PATH: config.get(CONF_FILE_PATH),
-    }
+    _LOGGER.debug("hass.data[%s] = %s", DOMAIN, hass.data[DOMAIN])
 
     hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, kill_raspistill)
 
-    file_path = setup_config[CONF_FILE_PATH]
+    file_path = hass.data[DOMAIN][CONF_FILE_PATH]
```

Comment:
Revert these changes.

Reply:
I don't really understand why.
I believe it's better to keep hass.data up to date so it has correct conf values.
Isn't it ?

Reply:
We will store a reference to `hass.data[DOMAIN]` in `setup_config` so we don't need to repeat `hass.data[DOMAIN]` everywhere.